### ISSUE TITLE
Spout transformers

### ISSUE THREAD

**Issue by jtojnar** [SSilence/selfoss#858]
Currently there are some derived spouts like “RSS (with Readability)“ or “RSS (with FullTextRss)“. It would be nice if there were spout transformers that could modify the output of the spouts. Then it would be possible to fix #705 simply by applying FullTextRss transformer to Reddit spout.

Additionally, instead of just `map`, the transformers could implement `filterMap` thus allowing smarter filters for spouts.

**Comment by lyda** [comment]
Would this relate to #606?

**Comment by jtojnar** [comment]
@lyda Yes. The current filters only use a single regex. Spout transformers would allow setting multiple filters. Something like this:

![screenshot-2017-9-15 selfoss 2637](https://user-images.githubusercontent.com/705123/30482110-14f388d8-9a22-11e7-9e23-b0489d55f31e.png)

**Comment by jtojnar** [comment]
I initially wanted to add advanced filters as part of this but since it is a feature many people including me want, I decided to split it out into #1125.
